vent/coordinator/ipc.py

```diff
@@ -1,28 +1,90 @@
 import threading
 import time
+from enum import Enum, auto
+import pickle
+import socket
+
+buffer_size = 65536
+
+
+class IPCCommand(Enum):
+    START = auto()
+    STOP = auto()
+    GET_SENSORS = auto()
+    GET_CONTROL = auto()
+    SET_CONTROL = auto()
+    GET_ACTIVE_ALARMS = auto()
+    GET_LOGGED_ALARMS = auto()
+    CLEAR_LOGGED_ALARMS = auto()
+
+
+class IPCMessage:
+    def __init__(self, command, args=None):
+        """
+        :param command: ENUM in IPCCommand
+        """
+        self.command = command
+        # TODO: current only assume one arg
+        self.args = args
 
 
 class IPC:
     # Class for communicating between processes either by sockets
     # or named-pipes
     #   Functions:
-    def __init__(self, listen=False):
+    def __init__(self, listen, addr=None, port=None):
+        self.listen = listen
+        self.s = socket.socket()
+        if listen:
+            # TODO: can start random port on testing, but deterministic port on deployment
+            self.s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            self.s.bind(('', 0))
+            self.addr, self.port = self.s.getsockname()
+            # print(self.addr)
+            # print(self.port)
+            # print('listen socket created')
+            self.conn = None
+            self.connection_thread = threading.Thread(target=self.connect, daemon=True)
```

Comment:
As written, `self.conn` is set to None immediately after init and there is nothing to stop `send_msg` or other methods that use it from being called. Need a flag that indicates when the connection is successful, and other messages need to check for it.